xain_fl/fl/coordinator/controller.py

```diff
@@ -7,56 +7,60 @@
 import numpy as np
 
 
+# TODO: raise exceptions for invalid attribute values: https://xainag.atlassian.net/browse/XP-387
 class Controller(ABC):
     """Abstract base class which provides an interface to the coordinator that
     enables different selection strategies.
 
     Attributes:
-        participant_ids (:obj:`list` of :obj:`str`): The list of IDs of the
-            all the available participants, a subset of which will be selected.
         fraction_of_participants (:obj:`float`, optional): The fraction of total
             participant ids to be selected. Defaults to 1.0, meaning that
-            all participant ids will be selected.
+            all participant ids will be selected. It must be in the (0.0, 1.0] interval.
     """
 
-    def __init__(
-        self, participant_ids: List[str], fraction_of_participants: float = 1.0
-    ) -> None:
-        self.participant_ids: List[str] = participant_ids
+    def __init__(self, fraction_of_participants: float = 1.0) -> None:
         self.fraction_of_participants: float = fraction_of_participants
-        self.num_ids_to_select: int = self.get_num_ids_to_select()
 
-    def get_num_ids_to_select(self) -> int:
+    def get_num_ids_to_select(self, participant_ids: List[str]) -> int:
```

Comment:
Is it necessary to pass the whole list of ids to the method instead of just passing the total number of ids?

Reply:
good catch. Might be an option to calculate the length of participant ids first for example in the `select_participant_ids_and_init_round` method in `coordinator.py`

Reply:
makes sense! went with @Robert-Steiner fix on line 65 (see comment below). will be part of next commit